server/index.ts

```diff
@@ -55,22 +55,61 @@ import {
 } from "../build/blog.js";
 import { findCurriculumPageBySlug } from "../build/curriculum.js";
 
-async function buildDocumentFromURL(url: string) {
-  const document = Document.findByURL(url);
-  if (!document) {
-    return null;
+async function buildDocumentFromURL(res, url: string) {
+  try {
+    console.time(`buildDocumentFromURL(${url})`);
+    const document = Document.findByURL(url);
+    if (!document) {
+      return null;
+    }
+    const documentOptions = {};
+    if (CONTENT_TRANSLATED_ROOT) {
+      // When you're running the dev server and build documents
+      // every time a URL is requested, you won't have had the chance to do
+      // the phase that happens when you do a regular `yarn build`.
+      document.translations = findTranslations(
+        document.metadata.slug,
+        document.metadata.locale
+      );
+    }
+    const built = await buildDocument(document, documentOptions);
+
+    if (built) {
+      return { doc: built?.doc, url };
```

Comment:
This breaks things in `app.get("/*/docs/*/index.json"` and `app.get("/*/docs/*/metadata.json"`, would it be simpler to only wrap things in `app.get("/*/docs/*"`, rather than also unwrap in the other two routes?